File: src/main/java/me/conorthedev/mediamod/command/MediaModCommand.java
package me.conorthedev.mediamod.command;

import mcp.MethodsReturnNonnullByDefault;
import me.conorthedev.mediamod.gui.GuiMediaModSettings;
import me.conorthedev.mediamod.util.TickScheduler;
import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;
import net.minecraftforge.fml.client.FMLClientHandler;

import javax.annotation.ParametersAreNonnullByDefault;
import java.util.Arrays;
import java.util.List;

/**
 * The client-side command to open the MediaMod GUI
 *
 * @see net.minecraft.command.ICommand
 */
@MethodsReturnNonnullByDefault
@ParametersAreNonnullByDefault
public class MediaModCommand extends CommandBase {

    @Override
    public String getName() {
        return "mediamod";
    }

    @Override
    public String getUsage(ICommandSender sender) {
        return "/mediamod";
    }

    @Override
    public List<String> getAliases() {
        return Arrays.asList("media", "mm");
    }

    @Override
    public void execute(MinecraftServer server, ICommandSender sender, String[] args) {
        TickScheduler.INSTANCE.schedule(1, () -> FMLClientHandler.instance().getClient().displayGuiScreen(new GuiMediaModSettings()));
    }

    @Override
    public boolean checkPermission(MinecraftServer server, ICommandSender sender) {
        return true;
    }

    @Override
    public int getRequiredPermissionLevel() {
        return -1;
    }
}

File: src/main/kotlin/com/mediamod/command/MediaModCommand.kt
/*
 *     MediaMod is a mod for Minecraft which displays information about your current track in-game
 *     Copyright (C) 2021 Conor Byrne
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */


package com.mediamod.command

import com.mediamod.gui.MediaModHomeGui
import com.mediamod.util.TickScheduler
import net.minecraft.client.Minecraft
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender

/**
 * The command for MediaMod to open the GUI
 */
class MediaModCommand : CommandBase() {
    override fun getCommandName() = "mediamod"
    override fun getCommandUsage(sender: ICommandSender?) = "/mediamod"
    override fun getRequiredPermissionLevel() = 0

    override fun processCommand(sender: ICommandSender?, args: Array<out String>?) {
        TickScheduler.schedule(1) {
            Minecraft.getMinecraft().displayGuiScreen(
                MediaModHomeGui()
            )
        }
    }
}
